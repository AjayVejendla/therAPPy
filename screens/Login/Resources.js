
import React, {Component} from 'react';
import {
    StyleSheet,
    Text,
    TextInput,
    TouchableOpacity,
    View,
    Linking,
} from 'react-native';

const links = [
    {
        id: 1,
        title: 'Rutgers CAPS website',
        link: 'http://health.rutgers.edu/medical-counseling-services/counseling/',
        description: 'Offers Counseling, Alcohol and Other Drug Assistance Program & Psychiatric Services',
    },
    {
        id: 2,
        title: 'US SAMHSA website',
        link: 'https://www.samhsa.gov/find-treatment',
        description: 'The US Government Substance Abuse and Mental Health Services Administration.',
    },
    {
        id: 3,
        title: 'US NIMH website',
        link: 'https://www.nimh.nih.gov/health/find-help/index.shtml',
        description: 'The US Government National Institute of Mental Health, guide to finding help.',
    },
];

const links2 =[
    {
        id: 1,
        title: 'Overview of Plutchik\'s Wheel of Emotions and its uses',
        link: 'https://positivepsychology.com/emotion-wheel/',
        description: 'Offers Counseling, Alcohol and Other Drug Assistance Program & Psychiatric Services',
    },
    {
        id: 2,
        title: 'Mobile Apps for mood tracking',
        link: 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5977660/',
        description: 'An analysis of features and user reviews',
    },
    {
        id: 3,
        title: 'CCAPS Survey Overview',
        link: 'https://drive.google.com/file/d/1mBaA0IObqsVCbnbqf4f6mKHWaR-b4d9S/view?usp=sharing',
        description: 'The Counseling Center Assessment of Psychological Symptoms',
    }
]

export default class Resources extends Component {
    render() {
        let resourcesArr;
        resourcesArr = links.map((obj,index) => (
            <View key={index} style={styles.minContainer}>
                <Text
                    style={styles.bigger}
                    onPress={() => Linking.openURL(obj.link)}>
                    {obj.title}
                </Text>
                <Text style={styles.loginText}>
                    {obj.description}
                </Text>
            </View>
        ));
        let researchArr;
        researchArr = links2.map((obj,index) => (
            <View key={index} style={styles.minContainer}>
                <Text
                    style={styles.bigger}
                    onPress={() => Linking.openURL(obj.link)}>
                    {obj.title}
                </Text>
                <Text style={styles.loginText}>
                    {obj.description}
                </Text>
            </View>
        ));
        return (
            <View style={styles.bigContainer}>
                <View style={styles.container}>
                    <Text style={styles.logo}>Resources</Text>
                    {resourcesArr}
                    <Text style={styles.logo}>Research</Text>
                    {researchArr}
                </View>
                <TouchableOpacity
                    style={styles.loginBtn}
                    onPress={() => {
                        this.props.navigation.navigate('BottomTabs')}}>
                    <Text style={styles.loginText}>To Dashboard</Text>
                </TouchableOpacity>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    bigContainer: {
        flex: 1,
        //therappy1: #707590
        //therappylessblue: #707586
        backgroundColor: '#f2e9e1',
        alignItems: 'center',
        justifyContent: 'flex-start',
    },
    container: {
        flex: 6,
        //therappy1: #707590
        //therappylessblue: #707586
        backgroundColor: '#f2e9e1',
        alignItems: 'center',
        justifyContent: 'space-evenly',
    },
    minContainer: {
        justifyContent: 'center',
        marginLeft: 20,
        marginRight: 20,
    },
    logo: {
        fontWeight: 'bold',
        fontSize: 50,
        //therappy1: #584d9b
        //therappylessblue: #584d9b
        //therappyteal: #20c0b0
        //therappyAIAIAIAI: #fdfdfb
        //lime green: #cbe86b
        color: '#7f58ff',
    },
    bigger: {
        fontWeight: 'bold',
        fontSize: 25,
        color: 'blue',
        textDecorationLine: 'underline',
    },
    inputView: {
        width: '80%',
        backgroundColor: '#ffffff',
        borderRadius: 25,
        height: 50,
        marginBottom: 20,
        justifyContent: 'center',
        padding: 20,
    },
    inputText: {
        height: 50,
        color: '#000000',
    },
    forgot: {
        color: 'black',
        fontSize: 14,
    },
    loginText: {
        color: 'black',
        fontSize: 18,
        marginLeft: 10,
        marginRight: 10,
    },
    rightBtn: {
        width: '45%',
        backgroundColor: '#ff666b',
        borderRadius: 10,
        height: 40,
        alignItems: 'center',
        justifyContent: 'center',
    },
    rightView: {
        flex: 1,
        marginTop: 25,
    },
    leftView: {
        flex: 1,
        marginLeft: 90,
        marginTop: 25,
    },
    leftBtn: {
        width: '45%',
        backgroundColor: '#ff666b',
        borderRadius: 10,
        height: 40,
        alignItems: 'center',
        justifyContent: 'center',
    },
});
