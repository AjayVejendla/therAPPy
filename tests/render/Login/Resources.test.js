import React from "react";
import Resources from "../../../screens/Login/Resources";
import renderer from "react-test-renderer";

test("Resources render test", () => {
  const tree = renderer.create(<Resources />).toJSON();
  expect(tree).toMatchSnapshot();
});
