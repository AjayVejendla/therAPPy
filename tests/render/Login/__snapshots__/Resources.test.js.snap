// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resources render test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f2e9e1",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2e9e1",
        "flex": 6,
        "justifyContent": "space-evenly",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#7f58ff",
          "fontSize": 50,
          "fontWeight": "bold",
        }
      }
    >
      Resources
    </Text>
    <View
      style={
        Object {
          "justifyContent": "center",
          "marginLeft": 20,
          "marginRight": 20,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "blue",
            "fontSize": 25,
            "fontWeight": "bold",
            "textDecorationLine": "underline",
          }
        }
      >
        Rutgers CAPS website
      </Text>
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 18,
            "marginBottom": 30,
            "marginLeft": 10,
            "marginRight": 10,
          }
        }
      >
        Offers Counseling, Alcohol and Other Drug Assistance Program & Psychiatric Services
      </Text>
    </View>
    <View
      style={
        Object {
          "justifyContent": "center",
          "marginLeft": 20,
          "marginRight": 20,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "blue",
            "fontSize": 25,
            "fontWeight": "bold",
            "textDecorationLine": "underline",
          }
        }
      >
        US SAMHSA website
      </Text>
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 18,
            "marginBottom": 30,
            "marginLeft": 10,
            "marginRight": 10,
          }
        }
      >
        The US Government Substance Abuse and Mental Health Services Administration.
      </Text>
    </View>
    <View
      style={
        Object {
          "justifyContent": "center",
          "marginLeft": 20,
          "marginRight": 20,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "blue",
            "fontSize": 25,
            "fontWeight": "bold",
            "textDecorationLine": "underline",
          }
        }
      >
        US NIMH website
      </Text>
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 18,
            "marginBottom": 30,
            "marginLeft": 10,
            "marginRight": 10,
          }
        }
      >
        The US Government National Institute of Mental Health, guide to finding help.
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#7f58ff",
          "fontSize": 50,
          "fontWeight": "bold",
        }
      }
    >
      Research
    </Text>
    <View
      style={
        Object {
          "justifyContent": "center",
          "marginLeft": 20,
          "marginRight": 20,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "blue",
            "fontSize": 25,
            "fontWeight": "bold",
            "textDecorationLine": "underline",
          }
        }
      >
        Overview of Plutchik's Wheel of Emotions and its uses
      </Text>
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 18,
            "marginBottom": 30,
            "marginLeft": 10,
            "marginRight": 10,
          }
        }
      >
        Offers Counseling, Alcohol and Other Drug Assistance Program & Psychiatric Services
      </Text>
    </View>
    <View
      style={
        Object {
          "justifyContent": "center",
          "marginLeft": 20,
          "marginRight": 20,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "blue",
            "fontSize": 25,
            "fontWeight": "bold",
            "textDecorationLine": "underline",
          }
        }
      >
        Mobile Apps for mood tracking
      </Text>
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 18,
            "marginBottom": 30,
            "marginLeft": 10,
            "marginRight": 10,
          }
        }
      >
        An analysis of features and user reviews
      </Text>
    </View>
    <View
      style={
        Object {
          "justifyContent": "center",
          "marginLeft": 20,
          "marginRight": 20,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "blue",
            "fontSize": 25,
            "fontWeight": "bold",
            "textDecorationLine": "underline",
          }
        }
      >
        CCAPS Survey Overview
      </Text>
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 18,
            "marginBottom": 30,
            "marginLeft": 10,
            "marginRight": 10,
          }
        }
      >
        The Counseling Center Assessment of Psychological Symptoms
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 18,
          "marginBottom": 30,
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      To Dashboard
    </Text>
  </View>
</View>
`;
